Source/GMCAbilitySystem/Public/Utility/GMASBoundQueue.h

```diff
@@ -0,0 +1,582 @@
+﻿// Fill out your copyright notice in the Description page of Project Settings.
+
+#pragma once
+
+#include "CoreMinimal.h"
+#include "GameplayTagContainer.h"
+#include "GMCMovementUtilityComponent.h"
+#include "InstancedStruct.h"
+#include "UObject/Object.h"
+#include "UObject/ConstructorHelpers.h"
+#include "GMASBoundQueue.generated.h"
+
+UENUM(BlueprintType)
+enum class EGMASBoundQueueOperationType : uint8
+{
+	None,
+	Add,
+	Remove,
+	Activate,
+	Cancel
+};
+
+USTRUCT(BlueprintType)
+struct FGMASBoundQueueOperationIdSet
+{
+	GENERATED_BODY()
+
+	UPROPERTY()
+	TArray<int> Ids = {};
+};
+
+USTRUCT(BlueprintType)
+struct GMCABILITYSYSTEM_API FGMASBoundQueueRPCHeader
+{
+	GENERATED_BODY()
+
+	UPROPERTY()
+	int32 OperationId { -1 };
+
+	UPROPERTY()
+	uint8 OperationTypeRaw { 0 };
+
+	EGMASBoundQueueOperationType GetOperationType() const
+	{
+		return static_cast<EGMASBoundQueueOperationType>(OperationTypeRaw);
+	}
+	
+	UPROPERTY()
+	FGameplayTag Tag { FGameplayTag::EmptyTag };
+
+	UPROPERTY()
+	FName ItemClassName { NAME_None };
+
+	UPROPERTY()
+	FGMASBoundQueueOperationIdSet PayloadIds {};
+
+	UPROPERTY()
+	FInstancedStruct InstancedPayload {};
+
+	UPROPERTY()
+	float RPCGracePeriodSeconds { 1.f };
+
+	UPROPERTY()
+	uint8 ExtraFlags { 0 };
+};
+
+template<typename C, typename T>
+struct GMCABILITYSYSTEM_API TGMASBoundQueueOperation
+{
+	FGMASBoundQueueRPCHeader Header {};
+	
+	// An actual class to be utilized with this, in case we need to instance it.
+	TSubclassOf<C> ItemClass { nullptr };
+
+	FInstancedStruct InstancedPayloadIds;
+	
+	// The struct payload for this item.
+	T Payload;	
+
+	// The instanced struct representation of this payload, used to actually
+	// bind for replication.
+	FInstancedStruct InstancedPayload;
+
+	EGMASBoundQueueOperationType GetOperationType() const
+	{
+		return Header.GetOperationType();
+	}
+
+	void SetOperationType(EGMASBoundQueueOperationType OperationType)
+	{
+		Header.OperationTypeRaw = static_cast<uint8>(OperationType);
+	}
+
+	int32 GetOperationId() const { return Header.OperationId; }
+
+	TArray<int32> GetPayloadIds() const { return Header.PayloadIds.Ids; }
+
+	FGameplayTag GetTag() const { return Header.Tag; }
+
+	bool GracePeriodExpired() const
```

Comment:
Minor pet peeve, mostly Epic's coding convention which has bool getters be a question, like `HasGracePeriodExpired()`.

Reply:
Fair, and easy enough to change.